File: python/vedant_bhatnagar/DevsnestProblems_THA/Phase_2/Trees/SymmetricTree.py
def f(p,q):
    if (p and not q) or (q and not p):
        return 0
    if not p and not q:
        return 1
    else:
        return 1 if p.val == q.val and f(p.left, q.right) and f(p.right, q.left) else 0


def solve(root):
    # CODE HERE
    if root:
        return f(root.left, root.right)

File: java/vansh_jain/trees/SymmetricTree.java
static int solve(TreeNode root){
    //CODE HERE 
    if(isSymmetric(root.left,root.right))
    return 1;
    return 0;
    }
    
static boolean isSymmetric(TreeNode left, TreeNode right){

    if(left==null || right==null)
        return left==right;
    return isSymmetric(left.left,right.right) && isSymmetric(left.right,right.left);

}